### ISSUE TITLE
Axiom Bug with Arm64

### ISSUE THREAD

**Issue by FireBall1725** [Moulberry/AxiomDocumentation#6]
Not sure where you want bugs for Axiom but I am unable to get Axiom to load on my computer, if there is a better place for this to be posted please let me know.

This is the crash report I get when Minecraft starts
```
java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'axiom'!
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
	at net.minecraft.class_310.<init>(class_310.java:458)
	at net.minecraft.client.main.Main.main(Main.java:211)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:468)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.UnsatisfiedLinkError: Can't load library: /var/folders/dw/8c_fht7n0ydgvrj_r3p9hw_40000gn/T/imgui-java-natives_1693826823853/libimgui-java64.dylib
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2393)
	at java.base/java.lang.Runtime.load0(Runtime.java:755)
	at java.base/java.lang.System.load(System.java:1953)
	at imgui.ImGui.<clinit>(ImGui.java:57)
	at com.moulberry.axiom.editor.ImGuiHelper.popAllStyleColors(ImGuiHelper.java:705)
	at com.moulberry.axiom.configuration.InternalConfiguration.save(InternalConfiguration.java:49)
	at com.moulberry.axiom.configuration.Configuration.save(Configuration.java:94)
	at com.moulberry.axiom.configuration.Configuration.load(Configuration.java:65)
	at com.moulberry.axiom.Axiom.onInitialize(Axiom.java:69)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)
	... 7 more
```

My guess is it should be trying to use the `libimgui-javaarm64.dylib` since that is the library file i have since i'm on an Arm64 processor.

Here is the entire log with system info / other mods loaded:
```
---- Minecraft Crash Report ----
// I feel sad now :(

Time: 2023-09-04 07:27:05
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'axiom'!
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
	at net.minecraft.class_310.<init>(class_310.java:458)
	at net.minecraft.client.main.Main.main(Main.java:211)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:468)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.UnsatisfiedLinkError: Can't load library: /var/folders/dw/8c_fht7n0ydgvrj_r3p9hw_40000gn/T/imgui-java-natives_1693826823853/libimgui-java64.dylib
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2393)
	at java.base/java.lang.Runtime.load0(Runtime.java:755)
	at java.base/java.lang.System.load(System.java:1953)
	at imgui.ImGui.<clinit>(ImGui.java:57)
	at com.moulberry.axiom.editor.ImGuiHelper.popAllStyleColors(ImGuiHelper.java:705)
	at com.moulberry.axiom.configuration.InternalConfiguration.save(InternalConfiguration.java:49)
	at com.moulberry.axiom.configuration.Configuration.save(Configuration.java:94)
	at com.moulberry.axiom.configuration.Configuration.load(Configuration.java:65)
	at com.moulberry.axiom.Axiom.onInitialize(Axiom.java:69)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)
	... 7 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
	at net.minecraft.class_310.<init>(class_310.java:458)

-- Initialization --
Details:
	Modules: 
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:211)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:468)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- System Details --
Details:
	Minecraft Version: 1.20.1
	Minecraft Version ID: 1.20.1
	Operating System: Mac OS X (aarch64) version 13.4.1
	Java Version: 17.0.3, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 553362264 bytes (527 MiB) / 1140850688 bytes (1088 MiB) up to 12884901888 bytes (12288 MiB)
	CPUs: 10
	Processor Vendor: Apple Inc.
	Processor Name: Apple M1 Max
	Identifier: Apple Inc. Family 0x1b588bb3 Model 0 Stepping 0
	Microarchitecture: ARM64 SoC: Firestorm + Icestorm
	Frequency (GHz): 3.23
	Number of physical packages: 1
	Number of physical CPUs: 10
	Number of logical CPUs: 10
	Graphics card #0 name: Apple M1 Max
	Graphics card #0 vendor: Apple (0x106b)
	Graphics card #0 VRAM (MB): 0.00
	Graphics card #0 deviceId: unknown
	Graphics card #0 versionInfo: unknown
	Memory slot #0 capacity (MB): 0.00
	Memory slot #0 clockSpeed (GHz): 0.00
	Memory slot #0 type: unknown
	Virtual memory max (MB): 35840.00
	Virtual memory used (MB): 24636.42
	Swap memory total (MB): 3072.00
	Swap memory used (MB): 2009.63
	JVM Flags: 7 total; -Xmx12G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		animatica: Animatica 0.6+1.20
		appleskin: AppleSkin 2.5.0+mc1.20
		architectury: Architectury 9.1.12
		authme: Auth Me 7.0.2+1.20
		autoclicker-fabric: Auto Clicker 1.20.1-1.8.0-build.25
		axiom: Axiom 1.1.1
			com_github_llamalad7_mixinextras: MixinExtras 0.2.0-beta.9
			com_moulberry_axiomclientapi_axiomclientapi_: axiomclientapi 1.0.0
		badpackets: Bad Packets 0.4.2
		beautifiedchatclient: Beautified Chat Client 2.2
		betterf3: BetterF3 7.0.1
		betterpingdisplay: Better Ping Display 1.1.1
		betterstats: Better Statistics Screen 2.10.3+1.20.1
			tcdcommons: TCD Commons API 2.10.2+1.20.1
		blur: Blur (Fabric) 3.1.0
			satin: Satin 1.13.0
		chat_heads: Chat Heads 0.10.20
		citresewn: CIT Resewn 1.1.3+1.20
			citresewn-defaults: CIT Resewn: Defaults 1.1.3+1.20
		clickthrough: ClickThrough 1.20-fabric0.83.0-0.4.1
			crowdin-translate: CrowdinTranslate 1.4+1.19.3
			gbfabrictools: GBfabrictools 1.4+1.20
		cloth-config: Cloth Config v11 11.0.99
			cloth-basic-math: cloth-basic-math 0.6.1
		collective: Collective 6.65
		controlling: Controlling For Fabric 12.0.1
		cupboard: cupboard 1.20.1-1.8
		enhancedblockentities: Enhanced Block Entities 0.9+1.20
			advanced_runtime_resource_pack: Runtime Resource Pack 0.6.7
			spruceui: SpruceUI 5.0.0+1.20
		fabric-api: Fabric API 0.84.0+1.20.1
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.34+4d8536c977
			fabric-biome-api-v1: Fabric Biome API (v1) 13.0.10+b3afc78b77
			fabric-block-api-v1: Fabric Block API (v1) 1.0.9+e022e5d177
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.39+b3afc78b77
			fabric-client-tags-api-v1: Fabric Client Tags 1.0.20+b3afc78b77
			fabric-command-api-v1: Fabric Command API (v1) 1.2.32+f71b366f77
			fabric-command-api-v2: Fabric Command API (v2) 2.2.11+b3afc78b77
			fabric-commands-v0: Fabric Commands (v0) 0.2.49+df3654b377
			fabric-containers-v0: Fabric Containers (v0) 0.1.61+df3654b377
			fabric-content-registries-v0: Fabric Content Registries (v0) 4.0.7+b3afc78b77
			fabric-convention-tags-v1: Fabric Convention Tags 1.5.3+b3afc78b77
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.18+aeb40ebe77
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 12.1.11+b3afc78b77
			fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.51+b3afc78b77
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.5.21+b3afc78b77
			fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.61+df3654b377
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.38+b04edc7a77
			fabric-item-group-api-v1: Fabric Item Group API (v1) 4.0.7+b3afc78b77
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.36+fb8d95da77
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.34+df3654b377
			fabric-loot-api-v2: Fabric Loot API (v2) 1.1.37+b3afc78b77
			fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.1.41+9e7660c677
			fabric-message-api-v1: Fabric Message API (v1) 5.1.6+b3afc78b77
			fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.47+b3afc78b77
			fabric-models-v0: Fabric Models (v0) 0.3.35+b3afc78b77
			fabric-networking-v0: Fabric Networking (v0) 0.3.48+df3654b377
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 11.0.6+b3afc78b77
			fabric-particles-v1: Fabric Particles (v1) 1.0.28+b3afc78b77
			fabric-recipe-api-v1: Fabric Recipe API (v1) 1.0.18+b3afc78b77
			fabric-renderer-api-v1: Fabric Renderer API (v1) 3.1.0+c154966e77
			fabric-renderer-indigo: Fabric Renderer - Indigo 1.4.0+c154966e77
			fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.44+df3654b377
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.33+b3afc78b77
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.0.26+b3afc78b77
			fabric-rendering-v0: Fabric Rendering (v0) 1.1.47+df3654b377
			fabric-rendering-v1: Fabric Rendering (v1) 3.0.6+b3afc78b77
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.3.5+ea08f9d877
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.27+b3afc78b77
			fabric-sound-api-v1: Fabric Sound API (v1) 1.0.12+b3afc78b77
			fabric-transfer-api-v1: Fabric Transfer API (v1) 3.2.2+b3afc78b77
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 4.2.0+b3afc78b77
		fabric-language-kotlin: Fabric Language Kotlin 1.10.10+kotlin.1.9.10
			org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.9.10
			org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.9.10
			org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.9.10
			org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.9.10
			org_jetbrains_kotlinx_atomicfu-jvm: atomicfu-jvm 0.22.0
			org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.7.3
			org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.7.3
			org_jetbrains_kotlinx_kotlinx-datetime-jvm: kotlinx-datetime-jvm 0.4.0
			org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm: kotlinx-serialization-cbor-jvm 1.6.0
			org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.6.0
			org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.6.0
		fabricloader: Fabric Loader 0.14.21
		fallingleaves: Falling Leaves 1.15.2+1.20.1
		farsight: Farsight Mod 1.20.1-4.1
			org_jctools_jctools-core: jctools-core 4.0.1
		forgeconfigapiport: Forge Config API Port 8.0.0
		freecam: Freecam 1.2.0+1.20
		fullbrightnesstoggle: Full Brightness Toggle 3.0
		indium: Indium 1.0.25+mc1.20.1
		inventoryhud: Inventory HUD + 3.4.18
		iris: Iris 1.6.8
			io_github_douira_glsl-transformer: glsl-transformer 2.0.0-pre13
			org_anarres_jcpp: jcpp 1.4.14
			org_antlr_antlr4-runtime: antlr4-runtime 4.11.1
		jade: Jade 11.5.1
		java: OpenJDK 64-Bit Server VM 17
		journeymap: Journeymap 5.9.10
			journeymap-api-fabric: JourneyMap API 1.20-1.9-fabric-SNAPSHOT
		litematica: Litematica 0.15.0
		litematica_printer: Litematica Printer 3.2
		malilib: MaLiLib 0.16.1
		midnightlib: MidnightLib 1.4.1
		minecraft: Minecraft 1.20.1
		minihud: MiniHUD 0.27.0
		modmenu: Mod Menu 7.2.1
		morechathistory: MoreChatHistory 1.1.1
		mousetweaks: Mouse Tweaks 2.25
		mousewheelie: Mouse Wheelie 1.12.2+mc1.20.1
			amecsapi: Amecs API 1.5.1+mc1.20-pre1
			coat: Coat 1.0.0-beta.20+mc1.20-pre1
			fabric-api-base: Fabric API Base 0.4.30+7abfd51577
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.6.1+e91849a877
			fabric-item-api-v1: Fabric Item API (v1) 2.1.27+b3afc78b77
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.2.21+b3afc78b77
			fabric-networking-api-v1: Fabric Networking API (v1) 1.3.9+b3afc78b77
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 2.3.0+2e061fd477
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.11.9+132c48c177
			fabric-screen-api-v1: Fabric Screen API (v1) 2.0.7+b3afc78b77
			tweed4_annotated: tweed4_annotated 1.3.1+mc1.20-pre1
			tweed4_base: tweed4_base 1.7.1+mc1.20-pre1
			tweed4_data: tweed4_data 1.2.1+mc1.20-pre1
			tweed4_data_hjson: tweed4_data_hjson 1.1.1+mc1.20-pre1
			tweed4_tailor_coat: tweed4_tailor_coat 1.1.3+mc1.20-pre1
			tweed4_tailor_lang_json_descriptions: tweed4_tailor_lang_json_descriptions 1.1.0+mc1.20-pre1
			tweed4_tailor_screen: tweed4_tailor_screen 1.1.4+mc1.20-pre1
		optigui: OptiGUI 2.1.5
			optiglue: OptiGlue 2.1.5-mc.1.19.3
			org_apache_commons_commons-text: commons-text 1.10.0
			org_ini4j_ini4j: ini4j 0.5.4
		physicsmod: Physics Mod 3.0.6
		reeses-sodium-options: Reese's Sodium Options 1.6.5+mc1.20.1-build.95
		replaymod: Replay Mod 1.20.1-2.6.13
		roughlyenoughitems: Roughly Enough Items 12.0.652
			error_notifier: Error Notifier 1.0.9
		searchables: Searchables 1.0.1
		shulkerboxtooltip: Shulker Box Tooltip 4.0.4+1.20.1
		simpleshulkerpreview: Simple Shulker Preview 2.4.0
		skinlayers: 3d Skin Layers 1.5.4-mc1.20.1
		sodium: Sodium 0.5.2
		sodium-extra: Sodium Extra 0.5.1+mc1.20.1-build.112
			caffeineconfig: CaffeineConfig 1.3.0+1.17
		status: Status 1.20.1-1.0.4
		textutilities: Text Utilities 1.6.1+1.20.1
		totemcounter: TotemCounter 1.3.0+mc1.20
		ukulib: ukulib 0.7.0+1.20
			com_moandjiezana_toml_toml4j: toml4j 0.7.2
			gs_mclo_java_mclogs-java: mclogs-java 2.1.1
		voicechat: Simple Voice Chat 1.20.1-2.4.24
		worldedit: WorldEdit 7.2.15+6463-5ca4dff
		worldeditcui: WorldEditCUI 1.20+01
		yet_another_config_lib_v3: YetAnotherConfigLib 3.1.0+1.20
			com_twelvemonkeys_common_common-image: common-image 3.10.0-SNAPSHOT
			com_twelvemonkeys_common_common-io: common-io 3.10.0-SNAPSHOT
			com_twelvemonkeys_common_common-lang: common-lang 3.10.0-SNAPSHOT
			com_twelvemonkeys_imageio_imageio-core: imageio-core 3.10.0-SNAPSHOT
			com_twelvemonkeys_imageio_imageio-metadata: imageio-metadata 3.10.0-SNAPSHOT
			com_twelvemonkeys_imageio_imageio-webp: imageio-webp 3.10.0-SNAPSHOT
	Launched Version: fabric-loader-0.14.21-1.20.1
	Backend library: LWJGL version 3.3.1 SNAPSHOT
	Backend API: Unknown
	Window size: <not initialized>
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: <disabled>
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	CPU: <unknown>
```

**Comment by gauravsatish** [comment]
this is not the place for it, please go to discord.gg/axiomtool

**Comment by Moulberry** [comment]
Not the place, also was just fixed in 2.0.0
